### PR TITLE
Fix various messages and documentation bugs

### LINKED ISSUE #146: [PE-D][Tester E] No parameters were updated/edited but result displays Edited Person
I first entered add -n John Doe123 -p 98765432 -e johnd@example.com -a 311, Clementi Ave 2, #02-25 -t friends -t owesMoney

Then i entered edit 1 -n John Doe123

Though there were no actual changes, the resultDisplay is Edited Person: John Doe123; Phone: 98765432; Email: johnd@example.com; Address: 311, Clementi Ave 2, #02-25; Tags: [owesMoney][friends]

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: xinen26/ped#4

### LINKED ISSUE #150: [PE-D][Tester D] Error message for adding entire list of prof
## Steps to replicate 

1. Add one prof, e.g. `prof -n aaron`.

2. Add all prof with `prof -a`.

## Expected

Success message.

## Actual

Error message.

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: shunjieee/ped#2

### LINKED ISSUE #192: [PE-D][Tester E] Unique Booking Descriptions for Booking System in UG
Unlike what is stated in UG that Booking Description must be non-null and unique, the unique part actually does not really apply to me because I was able to run

book -n CS2101 Consult -s 2023-12-31 19:00 -e 2023-12-31 23:00 -note DO
HOMEWORK

book -n CS2101 Consult -s 2023-12-31 19:01 -e 2023-12-31 23:00 -note DO
HOMEWORK

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: xinen26/ped#15

### PR DESCRIPTION
**Pull request by jerremyng**
1. Closes #146
* Edit message will show `edited ...` no matter if any changes are made or not.
* Fix here was to check equality of Person objects and throw message if not changed

2. Closes #150
* Adding a professor would result in duplicate person message shown if there are duplicates
* This is incorrect as it does not show the user the status of the operation
* Lets update this to show how many profs added and how many duplicates
Screenshot 2024-04-11 at 5 14 47 PM

3. Closes #192 
* Trivial documentation bug

### CODE REVIEW

**Comment by cyqjoseph** [comment]
Looks good! But we may need to update the UG to reflect the new image shown

**Comment by cyqjoseph** [comment]
On second thoughts, this might be a feature addition as we are showing how many profs are going to be added

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -122,11 +122,25 @@ public boolean equals(Object other) {
...
+                || (this.email != null && this.email.equals(otherPerson.email));
+
+        boolean addressEquals = (this.address == null && otherPerson.address == null)
+                || (this.address != null && this.address.equals(otherPerson.address));
+
+        boolean tagsEquals = (this.tags == null && otherPerson.tags == null)
+                || (this.tags != null && this.tags.equals(otherPerson.tags));
+
+        // Return true if all attributes are equal
+        return nameEquals && phoneEquals && emailEquals && addressEquals && tagsEquals;
```
Comment (ChrysanthemumT) [inline src/main/java/seedu/address/model/person/Person.java:143]: great defensive programming here!

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 2]
```diff
@@ -52,6 +52,18 @@ public void execute_allFieldsSpecifiedUnfilteredList_success() {
...
+        // Test that editing first person without change leads to error
+        Person editedPerson = model.getFilteredPersonList().get(0);
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder(editedPerson).build();
+        EditCommand editCommand = new EditCommand(INDEX_FIRST_PERSON, descriptor);
+
+        String expectedMessage = String.format(EditCommand.MESSAGE_NOT_EDITED);
+
+        assertCommandFailure(editCommand, model, expectedMessage);
+    }
+
```
Comment (ChrysanthemumT) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:66]: Thanks for the test cases!

**Review by ChrysanthemumT — APPROVED** [review]
lgtm!

**Pull request merged**